liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java

```diff
@@ -576,8 +576,32 @@ private String getContextRoot(EnterpriseArchive ear, WebArchive war) throws Depl
 			   closeQuietly(input);
 		   }
 	   }
-	   return createDeploymentName(war.getName());
-	}
+	   return getContextRoot(war);
+   }
+
+   private String getContextRoot(WebArchive war) throws DeploymentException {
+       org.jboss.shrinkwrap.api.Node webExtXmlNode = war.get("WEB-INF/ibm-web-ext.xml");
+       if(webExtXmlNode != null && webExtXmlNode.getAsset() != null) {
+           InputStream input = null;
+           try {
+               input = war.get("WEB-INF/ibm-web-ext.xml").getAsset().openStream();
+               Document webExtXml = readXML(input);
+               XPath xPath = XPathFactory.newInstance().newXPath();
+               XPathExpression ctxRootSelector = xPath.compile("//context-root/@uri");
+               String ctxRoot = ctxRootSelector.evaluate(webExtXml);
+               if(ctxRoot != null && ctxRoot.trim().length() > 0) {
+                   return ctxRoot;
```

Comment:
should we `trim()` the context root here?

Reply:
It's an attribute, so I wouldn't expect spurious whitespace, so maybe not...

It just seemed odd to use `trim` in the check on the previous line but not for the returned value